addons/mail/controllers/mail.py

```diff
@@ -197,3 +201,58 @@ def mail_action_unfollow(self, model, res_id, pid, token, **kwargs):
             'model_name': request.env['ir.model'].sudo()._get(model).display_name,
             'access_url': record._notify_get_action_link('view', model=model, res_id=res_id) if display_link else False,
         })
+
+    @http.route("/mail/message", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def mail_message_redirect(self, message_id):
+        message = request.env['mail.message'].search([('id', '=', message_id)])
+        if not message:
+            return {'error': 'Unauthorized'}
```

Comment:
Shouldn't we raise an error instead?

Reply:
It is to handle this specific error this way. If we were to raise `Unauthorized` than It's a bit tricky on the client side to validate which error was raised.